File: backend/app/services/calendar.py
import httpx


async def create_event(access_token: str, title: str, start_datetime: str, end_datetime: str | None, location: str | None, description: str | None):
    headers = {"Authorization": f"Bearer {access_token}"}
    payload = {
        "summary": title,
        "start": {"dateTime": start_datetime},
        "end": {"dateTime": end_datetime or start_datetime},
        "location": location,
        "description": description,
    }
    async with httpx.AsyncClient(timeout=15) as client:
        resp = await client.post(
            "https://www.googleapis.com/calendar/v3/calendars/primary/events",
            headers=headers,
            json=payload,
        )
        resp.raise_for_status()
        return resp.json()

File: backend2/src/services/calendar.ts
export async function createEvent(
  accessToken: string,
  title: string,
  startDatetime: string,
  endDatetime: string | null,
  location: string | null,
  description: string | null
): Promise<any> {
  const response = await fetch("https://www.googleapis.com/calendar/v3/calendars/primary/events", {
    method: "POST",
    headers: {
      Authorization: `Bearer ${accessToken}`,
      "Content-Type": "application/json"
    },
    body: JSON.stringify({
      summary: title,
      start: { dateTime: startDatetime },
      end: { dateTime: endDatetime ?? startDatetime },
      location,
      description
    })
  });
  const text = await response.text();
  if (!response.ok) {
    const error = new Error(text || response.statusText);
    (error as any).status = response.status;
    (error as any).responseText = text;
    throw error;
  }
  return text ? JSON.parse(text) : {};
}
